src/Classes/ServiceAPI/MyRadio_Scheduler.php

```diff
@@ -96,22 +96,25 @@ public static function countPendingDisputes()
      *
      * @return int The new termid
      */
-    public static function addTerm($start, $descr)
+    public static function addTerm($start, $descr, $num_weeks)
     {
         if (date('D', $start) !== 'Mon') {
             throw new MyRadioException('Terms must start on a Monday.', 400);
         }
-
+        $num_weeks = (int)$num_weeks;
+        if (!is_int($num_weeks)){
+            throw new MyRadioException('Weeks must be an integer.', 400); 
+        }
         // Let's make this a GMT thing, exactly midnight
         $ts = gmdate('Y-m-d 00:00:00+00', $start);
-        $end = $start + (86400 * 68); // +68 days, to Friday of the 10th week
+        $end = $start + (86400 * ((7*$num_weeks)-2)); // to Friday of the 10th week
         $te = gmdate('Y-m-d 00:00:00+00', $end);
 
-        echo "INSERT INTO terms (start, finish, descr) VALUES ('$ts', '$te', '$descr') RETURNING termid";
+        echo "INSERT INTO terms (start, finish, descr, weeks) VALUES ('$ts', '$te', '$descr', '$num_weeks') RETURNING termid";
 
         return self::$db->fetchColumn(
-            'INSERT INTO terms (start, finish, descr) VALUES ($1, $2, $3) RETURNING termid',
-            [$ts, $te, $descr]
+            'INSERT INTO terms (start, finish, descr, weeks) VALUES ($1, $2, $3, $4) RETURNING termid',
```

Comment:
Is this a new column? If so, have you written the SQL to add it, and put it in `schema/patches`?

Reply:
I added it to the test db using the pgadmin ui. I have no idea what schema/patches is so I will need that explaining please I'm sorry for being stupid